src/drivers/QtTestDriver.cpp

```diff
@@ -8,25 +8,32 @@ namespace cucumber {
 namespace internal {
 
 const InvokeResult QtTestStep::invokeStepBody() {
-    QTemporaryFile file;
-    if (!file.open()) {
-        return InvokeResult::failure("Unable to open temporary file needed for this test");
+    QString file_name;
+    {
+        QTemporaryFile file;
+        if (!file.open()) {
+            return InvokeResult::failure("Unable to open temporary file needed for this test");
+        }
+        file.close();
```

Comment:
Closing the file here can lead to bugs. In the sequence between here and when the file is opened again to read, another process could create a file with the same name. Is it possible to open the file for reading while another process is writing to it?

Reply:
This is the major problem. The file MUST go out of scope (this means also implicitly closed()) for the test to have write access on windows.
Cause the filename as the process name as file name part, another process should not have the same name.
Considering multiple parallel tests of the same executable is an issue, could be fixed by adding the pid in the file name template.

Reply:
Ok, I see the problems. According to the documentation, it does indeed look like there is no easy solution.

As it gets more tricky a helper class may be needed. Something like (I haven't compiled nor test the code, needs also one comment to explain why we do it in such a complicated manner):
```C++

class TemporaryFileWrapper {
public:
    static std::optional<TemporaryFileWrapper> create()
    {
        QTemporaryFile tempFile{};
        tempFile.setAutoRemove(false);

        if(!tempFile.open()) {
            return {};
        }

        return {tempFile.fileName() + ".txt"};
    }

    TemporaryFileWrapper(QString filename): filename{filename} {
    }

    ~TemporaryFileWrapper() {
        QFile::remove(filename);
    }

    QString name() const {
        return filename;
    }

    QString read() const {
        QFile file{ filename };
        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
            return QString();
        QTextStream in(&file);
        return in.readAll();
    }

private:
    QString filename;
};


const InvokeResult QtTestStep::invokeStepBody() {
    const auto file = TemporaryFileWrapper::create();
    if (!file) {
        return InvokeResult::failure("Unable to open temporary file needed for this test");
    }
    
    QtTestObject testObject(this);
    int returnValue = QTest::qExec(
        &testObject,
        QStringList() << "test"
                      << "-o" << file->name()
    );
    
    if (returnValue == 0) {
        return InvokeResult::success();
    } else {
        return InvokeResult::failure(file->read().toLocal8Bit());
    }
}
```

Reply:
A solution could be to use the standard function tmpnam. So no file is generated, it is just the name generated.
The template should also be application name and pid. But this always gives me a warning cause considered insecure) which should be ignored.

Reply:
Saw your suggestion to late :smile: Would give it a try tomorrow.

Reply:
Tried to integrate the wrapper without std::optional.
Open Issues: if Autoremove is disabled, the temporary file is NOT removed with QFile::remove. Still existing.
The return value is true (So file should be removed) but is still "there" in the temp folder.
Autoremove in "normal" behaviour fixes this. Cause we do want only the filename, this would be ok for me.